File: 08/Java/TripleExpression/Square.java
package TripleExpression;

public class Square<T> implements TripleExpression<T> {
	private TripleExpression<T> value;
    private Operator<T> op;
    public Square(TripleExpression<T> val, Operator<T> op) {
        value = val;
        this.op = op;
    }

    public T evaluate(T x, T y, T z) throws Exception {
        T a = value.evaluate(x, y, z);
        return op.square(a);
    }
}

File: 07/CSharp/TripleExpressionApp/TripleExpressionApp/Square.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Square : ITripleExpression
    {
        private ITripleExpression value;
        public Square(ITripleExpression value)
        {
            this.value = value;
        }

        public double Evaluate(double x, double y, double z)
        {
            double a = value.Evaluate(x, y, z);
            return a * a;
        }
    }
}
